05/src/academy/pocu/comp2500samples/w05/baseentity/Program.java

```diff
@@ -0,0 +1,123 @@
+package academy.pocu.comp2500samples.w05.baseentity;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.UUID;
+
+public class Program {
+    public static void main(String[] args) {
+        User user1 = new User(
+                UUID.randomUUID(),
+                OffsetDateTime.now(ZoneOffset.UTC),
+                OffsetDateTime.now(ZoneOffset.UTC),
+                "Tom",
+                "Smith",
+                "tommy hammer");
+
+        printUserInformation(user1);
+
+        BaseEntity user2 = new User(
+                UUID.randomUUID(),
+                OffsetDateTime.now(ZoneOffset.UTC),
+                OffsetDateTime.now(ZoneOffset.UTC),
+                "Kevin",
+                "Park",
+                "KtotheP");
+
+        // Compile Error!
+        // printUserInformation(user2);
+
+        printUserInformation((User) user2);
+
+        ((User) user2).setNickname("KevinInThePark");

```

Comment:
((User)user2).

Reply:
This, unfortunately... is what IDE's code formatter does.